src/rgw/rgw_op.cc

```diff
@@ -3770,6 +3787,7 @@ void RGWPutObj::execute(optional_yield y)
 
     buf_to_hex((const unsigned char *)supplied_md5_bin, CEPH_CRYPTO_MD5_DIGESTSIZE, supplied_md5);
     ldpp_dout(this, 15) << "supplied_md5=" << supplied_md5 << dendl;
+    execute_span->set_log("log", std::string("supplied_md5= ") + supplied_md5);
```

Comment:
if we want to reuse the same string for `dout` as well as the log in the span, we should probably add the span as a parameter to `ldpp_dout()`, so that we compose the string once and use it in the two places.

Reply:
we should be careful not to log extraneous detail, especially if it requires string formatting. lets strive for minimal overhead when tracing is disabled

Reply:
idea is to reuse whatever is already done for `ldout`.
so we don't add any extra string stream cost.